ui/app/components/secret-edit.js

```diff
@@ -235,6 +238,27 @@ export default Component.extend(FocusOnInsertMixin, {
       set(this.modelForData, 'secretData', this.secretData.toJSON());
     },
 
+    handleWrappedClick() {
+      if (this.isV2) {
+        const { path } = this.model.versions.objectAt(this.model.currentVersion - 1);
+        this.store
+          .adapterFor('secret-v2-version')
+          .queryRecord(this.model.backend, path, this.model.currentVersion, { wrapTTL: 1800 })
```

Comment:
Using `this.modelForData` here would help too so you could pass the id instead of the different parts from the secret-v2 model.